source/FreeRTOS_ARP.c

```diff
@@ -328,6 +328,9 @@ static void vProcessARPPacketReply( ARPPacket_t * pxARPFrame,
 
     if( pxARPWaitingNetworkBuffer != NULL )
     {
+        /* MISRA C-2012 Rule 11.3 warns about casting pointer type to a different data type.
+         * To be able to access various predefined fields from a data buffer, this mapping is intentional. */
+        /* coverity[misra_c_2012_rule_11_3_violation] */
```

Comment:
The mapping is intentional, but what a reviewer might want to know that whether this will cause an out of bound access. I think we should reword it similar to the following:

```suggestion
        /* MISRA C-2012 Rule 11.3 warns about casting pointer type to a different data
         * type. To be able to access various predefined fields from a data buffer,
         * this mapping is intentional and has been validated to not exceed the buffer. */
        /* coverity[misra_c_2012_rule_11_3_violation] */
```

And change it in all the places...

Reply:
looks good, will change the wording.